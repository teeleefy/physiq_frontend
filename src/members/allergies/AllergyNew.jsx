import { Form, FormGroup, Input, Button, Label, Alert } from "reactstrap";
import { useState, useContext } from "react";
import {MemberContext} from "../../auth/UserContext.js";
import PhysiqApi from "../../Api.js";
import { useNavigate } from "react-router-dom";
import '../styles/Allergy.css'

function AllergyNew(){
    let navigate = useNavigate();
    const { currentMember } = useContext(MemberContext);
    const [formMessages, setFormMessages] = useState([]);
    const [updateSuccess, setUpdateSuccess] =useState();
    const [formData, setFormData] = useState({name: "", reaction: "", notes: ""});

    async function addAllergy() {
        try {
          let newAllergy= await PhysiqApi.addMemberAllergy(currentMember.id, formData);
          return { success: true };
        } catch (errors) {
          console.error("Add Allergy failed", errors);
          return { success: false, errors };
        }
      }

    async function handleSubmit(evt){
        evt.preventDefault();
        // console.log(formData);
        let result = await addAllergy();
        
        if(result.success){
        //    alert("Saved Changes!") 
           setFormMessages(['Allergy Added!'])
           setUpdateSuccess(true);
           navigate("..", { relative: "path"});
        }
        else{
            setFormMessages(result.errors);
            setUpdateSuccess(false);
        }
    };

  
    /** Update local state w/curr state of input elem */
  
    const handleChange = evt => {
      const { name, value }= evt.target;
      setFormData(fData => ({
        ...fData,
        [name]: value
      }));
    };

    return(
        <>
            <Form className="AllergyNew-Form m-4" >
            <h1 className="Allergy-h1">Add Allergy:</h1>
            <p className="text-secondary">Fields marked with <span className="text-danger">*</span> are required.</p>
            <FormGroup>
                <Label className="Allergy-label" for="name" >
                    <b>Allergy Name</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="name"
                name="name"
                value={formData.name}
                placeholder="Enter allergy name..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
            <FormGroup>
                <Label className="Allergy-label" for="reaction">
                    <b>Reaction</b><span className="text-danger">*</span>
                </Label>
                <Input
                id="reaction"
                name="reaction"
                value={formData.reaction}
                placeholder="Enter reaction..."
                type="text"
                onChange={handleChange}
                />
                </FormGroup>
            <FormGroup>
                <Label className="Allergy-label" for="notes">
                    <b>Notes</b>
                </Label>
                <Input
                id="notes"
                name="notes"
                value={formData.notes}
                placeholder="Enter notes..."
                type="text"
                onChange={handleChange}
                />
            </FormGroup>
                {formMessages.length
                    ? formMessages.map(msg => <Alert color={updateSuccess ? "success": "danger"}>{msg}</Alert>)
                    : null
                }
                
                <Button className="btn-dark" onClick={handleSubmit}>Add Allergy</Button>
            </Form>
        </>
    )}
    
export default AllergyNew;