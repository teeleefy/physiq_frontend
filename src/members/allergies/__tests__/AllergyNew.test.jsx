import {renderWithRouterAndMemberContext, screen} from '../../../../__tests__/test-tools';
import AllergyNew from '../AllergyNew';

describe('AllergyNew', () => {
    it('renders the AllergyNew component', () => {
      renderWithRouterAndMemberContext(<AllergyNew/>);
      screen.debug(); // prints out the jsx in the App component unto the command line
    });

    it("matches snapshot", function() {
      const { asFragment } = renderWithRouterAndMemberContext(<AllergyNew />);
      expect(asFragment()).toMatchSnapshot();
    });

  });
