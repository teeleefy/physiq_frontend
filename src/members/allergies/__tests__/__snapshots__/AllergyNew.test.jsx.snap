// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AllergyNew > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="AllergyNew-Form m-4"
  >
    <h1
      class="Allergy-h1"
    >
      Add Allergy:
    </h1>
    <p
      class="text-secondary"
    >
      Fields marked with 
      <span
        class="text-danger"
      >
        *
      </span>
       are required.
    </p>
    <div
      class="mb-3"
    >
      <label
        class="Allergy-label form-label"
        for="name"
      >
        <b>
          Allergy Name
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="name"
        maxlength="50"
        name="name"
        placeholder="Enter allergy name..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Allergy-label form-label"
        for="reaction"
      >
        <b>
          Reaction
        </b>
        <span
          class="text-danger"
        >
          *
        </span>
      </label>
      <input
        class="form-control"
        id="reaction"
        maxlength="100"
        name="reaction"
        placeholder="Enter reaction..."
        type="text"
        value=""
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="Allergy-label form-label"
        for="notes"
      >
        <b>
          Notes
        </b>
      </label>
      <textarea
        class="form-control"
        id="notes"
        maxlength="250"
        name="notes"
        placeholder="Enter notes..."
      />
    </div>
    <p
      class="text-secondary"
    >
      0/250 Characters
    </p>
    <button
      class="btn-dark btn btn-secondary"
      type="button"
    >
      Add Allergy
    </button>
  </form>
</DocumentFragment>
`;
